plone/app/discussion/browser/ban_integration.py

```diff
@@ -0,0 +1,194 @@
+"""Comment form integration for ban management."""
+
+from plone.app.discussion.ban import BAN_TYPE_COOLDOWN
+from plone.app.discussion.ban import BAN_TYPE_PERMANENT
+from plone.app.discussion.ban import can_user_comment
+from plone.app.discussion.ban import get_ban_manager
+from plone.app.discussion.ban import is_comment_visible
+from plone.app.discussion.interfaces import _
+from Products.CMFCore.utils import getToolByName
+from Products.statusmessages.interfaces import IStatusMessage
+
+
+try:
+    from plone.app.discussion.interfaces import IDiscussionSettings
+    from plone.registry.interfaces import IRegistry
+    from zope.component import getUtility
+except ImportError:
+    pass
+
+
+def check_user_ban_before_comment(comment_form, data):
+    """Check if user is banned before allowing comment submission.
+
+    This function should be called from the comment form's handleComment method.
+    Returns True if comment should be allowed, False otherwise.
+    """
+    context = comment_form.context
+    request = comment_form.request
+
+    # Check if ban system is enabled
+    try:
+        registry = getUtility(IRegistry)
+        settings = registry.forInterface(IDiscussionSettings, check=False)
+        if not getattr(settings, "ban_enabled", False):
+            return True  # Ban system disabled, allow comment
+    except Exception:
```

Comment:
Bare exceptions are not good, b/c they may shadow other problems. May you be more specific here?